### PR TITLE
refactor(terraform): improve readability and consistency in templates and variables

### PR DESCRIPTION
**Pull request by sachincool**
Enhance the readability of the README by aligning table columns and adding spacing.
Refactor locals.tf and templates to use consistent naming conventions,
such as changing 'template_file_map' to 'provider_template' and
'secret_store_enabled' to 'parameter_store_enabled'.
Simplify GCP configuration by consolidating service account data into
'sa_auth_data'. Update variable defaults and descriptions for clarity.
Remove unused 'cluster.yaml.tpl' to streamline the codebase.

### CODE REVIEW

File: variables.tf [hunk 1]
```diff
@@ -225,53 +225,50 @@ variable "gcp_sa_private_key" {
   default     = null
 }
 
-variable "gcp_sa_project_id" {
+# GCP Storage Configuration
+variable "gcp_artifact_registry_url" {
   type        = string
-  description = "GCP Project ID associated with the Service Account"
+  description = "URL for GCP Artifact Registry (e.g., 'LOCATION-docker.pkg.dev/PROJECT_ID/REPOSITORY')"
```
Comment (shubhamrai1993) [inline variables.tf:231]: Will the registry url contain the repository?

File: variables.tf [hunk 2]
```diff
@@ -225,53 +225,50 @@ variable "gcp_sa_private_key" {
...
-  default     = "https://accounts.google.com/o/oauth2/auth"
+  description = "URL for GCP Storage bucket (e.g., 'gs://bucket-name')"
+  default     = null
 }
 
-variable "gcp_sa_token_uri" {
-  type        = string
-  description = "GCP OAuth token URI"
-  default     = "https://oauth2.googleapis.com/token"
+variable "gcp_cluster_integration_enabled" {
```
Comment (shubhamrai1993) [inline variables.tf:241]: what does this variable mean?

Comment (sachincool) [reply variables.tf:241]: `cluster_integration_enabled` is used if you want to skip integration https://github.com/truefoundry/terraform-truefoundry-integrations/blob/main/templates/aws.json.tpl#L50  
this was something ocli also used to pass, by default its true.

File: variables.tf (hunk 1 above)
Comment (sachincool) [reply variables.tf]: no, you're right, `LOCATION-docker.pkg.dev/PROJECT_ID` 
sent the fix.

**Review by shubhamrai1993 — APPROVED** [review]

**Pull request merged**